src/app/initialisation/syncPoolService.js

```diff
@@ -43,14 +48,22 @@ function SyncPoolService($q, mediator, workorderSync, workflowSync, resultSync,
     promises.push(workflowSync.createManager());
     promises.push(messageSync.createManager());
     promises.push(resultSync.createManager());
-    return $q.all(promises).then(function(managers) {
-      var map = {};
-      managers.forEach(function(managerWrapper) {
-        map[managerWrapper.manager.datasetId] = managerWrapper;
+
+    //Initialising the sync managers for the required datasets.
+    return syncService.manage(config.datasetIds.workorders, {}, {}, config.syncOptions)
```

Comment:
I think having them as separate calls allows the application developer to decide when to initialise the data sets.

Also the config for all of the data set would become very verbose when we start adding collision handlers etc.